src/Root.cpp

```diff
@@ -607,7 +607,7 @@ bool cRoot::FindAndDoWithPlayer(const AString & a_PlayerName, cPlayerListCallbac
 			size_t Rating = RateCompareString (m_PlayerName, a_pPlayer->GetName());
 			if ((Rating > 0) && (Rating >= m_BestRating))
 			{
-				m_BestMatch = a_pPlayer;
+				m_BestMatch = a_pPlayer->GetName();
```

Comment:
Also using the ID results in the rather unintuitive situation where if two players with the same name are connected and one leaves whilst FindAndDoWithPlayer is executing, the call fails as despite there never being a point in time during the execution where there are no players with that name connected.

The linerization of the operations that results in that condition is:

1. Two players with name "Foo"  and IDs 1 and 2 connect
2. Start FindAndDoWithPlayer("Foo")
3. ForEachPlayer() locates player with ID 1.
4. Player with ID 1 disconnects.
5. DoWIthEntityByID Fails as no entity with ID 1.
6. FindAndDoWithPlayer("Foo") fails.

When the code uses the name it automatically fails over the the player with ID 2.